pkg/apis/v1beta1/ec2nodeclass_validation_cel_test.go

```diff
@@ -0,0 +1,576 @@
+			updatedHash := nodeClass.Hash()
+			Expect(hash).ToNot(Equal(updatedHash))
+		},
+			Entry("InstanceProfile Drift", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{Role: "role-2"}}),
+			Entry("UserData Drift", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{UserData: aws.String("userdata-test-2")}}),
+			Entry("Tags Drift", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{Tags: map[string]string{"keyTag-test-3": "valueTag-test-3"}}}),
+			Entry("MetadataOptions Drift", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{MetadataOptions: &v1beta1.MetadataOptions{HTTPEndpoint: aws.String("test-metadata-2")}}}),
+			Entry("BlockDeviceMappings Drift", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{BlockDeviceMappings: []*v1beta1.BlockDeviceMapping{{DeviceName: aws.String("map-device-test-3")}}}}),
+			Entry("Context Drift", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{Context: aws.String("context-2")}}),
+			Entry("DetailedMonitoring Drift", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{DetailedMonitoring: aws.Bool(true)}}),
+			Entry("AMIFamily Drift", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{AMIFamily: aws.String(v1alpha1.AMIFamilyBottlerocket)}}),
+		)
+		DescribeTable("should not change hash when slices are re-ordered", func(changes v1beta1.EC2NodeClass) {
+			hash := nodeClass.Hash()
+			Expect(mergo.Merge(nodeClass, changes, mergo.WithOverride)).To(Succeed())
+			updatedHash := nodeClass.Hash()
+			Expect(hash).To(Equal(updatedHash))
+		},
+			Entry("Reorder Tags", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{Tags: map[string]string{"keyTag-2": "valueTag-2", "keyTag-1": "valueTag-1"}}}),
+			Entry("Reorder BlockDeviceMapping", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{BlockDeviceMappings: []*v1beta1.BlockDeviceMapping{{DeviceName: aws.String("map-device-2")}, {DeviceName: aws.String("map-device-1")}}}}),
+		)
+		It("should not change hash when behavior/dynamic fields are updated", func() {
+			hash := nodeClass.Hash()
+
+			// Update a behavior/dynamic field
+			nodeClass.Spec.SubnetSelectorTerms = []v1beta1.SubnetSelectorTerm{
+				{
+					Tags: map[string]string{"subnet-test-key": "subnet-test-value"},
+				},
+			}
+			nodeClass.Spec.SecurityGroupSelectorTerms = []v1beta1.SecurityGroupSelectorTerm{
+				{
+					Tags: map[string]string{"sg-test-key": "sg-test-value"},
+				},
+			}
+			nodeClass.Spec.AMISelectorTerms = []v1beta1.AMISelectorTerm{
+				{
+					Tags: map[string]string{"ami-test-key": "ami-test-value"},
+				},
+			}
+			updatedHash := nodeClass.Hash()
+			Expect(hash).To(Equal(updatedHash))
+		})
+		It("should expect two provisioner with the same spec to have the same provisioner hash", func() {
+			otherNodeClass := test.EC2NodeClass(v1beta1.EC2NodeClass{
+				Spec: nodeClass.Spec,
+			})
+			Expect(nodeClass.Hash()).To(Equal(otherNodeClass.Hash()))
+		})
+	})
+	Context("BlockDeviceMappings", func() {
```

Comment:
Can we validate some things in here about the new blockDeviceMapping logic now that we are doing Regex here instead of what we were doing before